File: tm_867.rs
impl Solution {
    pub fn transpose(matrix: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let mut result = Vec::new();

        for i in 0..matrix[0].len() {
            let mut col = Vec::new();
            for j in 0..matrix.len() {
                col.push(matrix[j][i]);
            }
            result.push(col)
        }

        return result;
    }
}

File: js/tm_867.js
/**
 * @param {number[][]} matrix
 * @return {number[][]}
 */
var transpose = function (matrix) {
  const transposedArr = [];

  // need to use matrix[0] length to loop first array of two dimentional array, this is to prevent when first demention and second dimention are not equal
  for (let i = 0; i < matrix[0].length; i++) {
    const col = [];
    // need to use whole matrix length to create a new array with different length when first demention and second dimention are not equal
    for (let j = 0; j < matrix.length; j++) {
      col.push(matrix[j][i]);
    }
    transposedArr.push(col);
  }
  return transposedArr;
};

const result = transpose([
  [1, 2, 3],
  [4, 5, 6],
]);
console.log(result);
